core/runtime/v2/README.md

```diff
@@ -537,3 +537,9 @@ It works with standard protobufs and GRPC services as well as generating clients
 The only difference between grpc and ttrpc is the wire protocol.
 ttrpc removes the http stack in order to save memory and binary size to keep shims small.
 It is recommended to use ttrpc in your shim but grpc support is currently an experimental feature.
+
+### reap orphan process
+After this commit https://github.com/torvalds/linux/commit/c6c70f4455d1eda91065e93cc4f7eddf4499b105 (merged in kernel 4.11)
```

Comment:
As far as I know, the shim is not designed to reap the orphan process created by exec-init.
However, the kernel has bug and it's like bug used as a feature for someone.
I'm not sure it's good place to highlight tini or pod-level pid namespace here.

Reply:
Yeah, we can highlight shim doesn't have responsibility to reap any orphan processes created by exec-init. And it's just for runc-shim.

Reply:
sometimes user use different kernel  have different behaviors ,they don't know why. I think we can let user know why

Reply:
> Yeah, we can highlight shim doesn't have responsibility to reap any orphan processes created by exec-init. And it's just for runc-shim.
runc-shim has process to reap orphan process but  doesn't work after kernel 4.11.
https://github.com/containerd/containerd/blob/b40a912e7b42c47bf06068508fe948840a80730b/pkg/shim/shim_unix.go#L85

Reply:
The reaper is for init container or exec-init, not for the orphan process.

Reply:
> The reaper is for init container or exec-init, not for the orphan process.

you are right, shim will  reap orphan process created by exec-init before kernel 4.11.